### PR TITLE
feat(portal): upgrade to angular 16

### PR DESCRIPTION
**Pull request by sumitparakh**
Angular signals PR will be followed by this PR.
#150

### CODE REVIEW

File: packages/portal/src/app/shared/guards/auth-guard.service.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { Injectable } from "@angular/core";
-import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from "@angular/router";
+import { ActivatedRouteSnapshot, Router, RouterStateSnapshot } from "@angular/router";
 import { AuthService } from "../authentication/auth.service";
 
 @Injectable({
   providedIn: "root",
 })
-export class AuthGuard implements CanActivate {
+export class AuthGuard  {
```
Comment (manekinekko) [inline packages/portal/src/app/shared/guards/auth-guard.service.ts:8]: should not this be converted to a function?

**Review by manekinekko — CHANGES_REQUESTED** [review]

File: packages/portal/src/app/shared/guards/auth-guard.service.ts (hunk 1 above)
Comment (sumitparakh) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts:8]: Yes. Will update

Comment (anfibiacreativa) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts]: That's one of the possibilities. Yes.

**Review by anfibiacreativa — CHANGES_REQUESTED** [review]

File: packages/portal/src/app/shared/guards/auth-guard.service.ts (hunk 1 above)
Comment (sumitparakh) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts]: Great. Will do that.

Comment (sumitparakh) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts]: Hi. I have addressed review comments.

**Comment by sumitparakh** [comment]
Also fixing the broken css that was probably caused by automated import path update by angular for Angular material legacy modules

File: packages/portal/src/app/shared/guards/auth-guard.service.ts (hunk 1 above)
Comment (anfibiacreativa) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts]: Hi Sumit, yes, thank you. I have been trying to open the PR in codespaces unsuccessfully. Can you please set the target branch to `feat/develop-pr219`, thank you!

Comment (anfibiacreativa) [reply packages/portal/src/app/shared/guards/auth-guard.service.ts]: Thank you! I will merge to this branch and test.

**Pull request merged**